runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -36,10 +37,50 @@
 namespace Catalyst::Runtime {
 
 /**
- * @brief Global quantum device unique pointer.
+ * @brief Global quantum device unique pointer with internal linkage.
  */
 thread_local static std::unique_ptr<ExecutionContext> CTX = nullptr;
 
+/**
+ * @brief Thread local device pointer and unique ID with internal linkage.
+ */
+thread_local static QuantumDevice *RTD_PTR = nullptr;
+thread_local static size_t RTD_ID = 0;
+
+/**
+ * @brief Initialize the device instance and update the value of RTD_PTR and RTD_ID
+ * to the new initialized device pointer and unique id.
+ */
+[[nodiscard]] bool initDevice(ExecutionContext *ec, std::string_view rtd_lib,
+                              std::string_view rtd_name, std::string_view rtd_kwargs)
+{
+    auto &&[key, rtd_ptr] = ec->getDevice(rtd_lib, rtd_name, rtd_kwargs);
+    RTD_PTR = rtd_ptr.get();
+    RTD_ID = key;
+
+    return true;
+}
+
+/**
+ * @brief Release the active device instance.
+ */
+void releaseDevice(ExecutionContext *ec)
```

Comment:
I'm curious, if the execution context is global, why does it need to be passed in? Or are there potentially multiple instances of the execution context?

Reply:
I re-asked the question in my recent review, so I'm closing this one again.

In general, even if a comment seems resolved by recent changes, it's good to state why and how you think it is now resolved before closing.